pkg/store/store.go

```diff
@@ -176,12 +177,13 @@ func mutate(ctx context.Context, s storeLoader, builder qb.Builder, values ...in
 	return nil
 }
 
-func (ds delegatingStore) Check(ctx context.Context, table *typedef.Table, builder qb.Builder, values ...interface{}) error {
-	testRows, err := ds.testStore.load(ctx, builder, values)
+func (ds delegatingStore) Check(ctx context.Context, syncer *CheckSyncer, table *typedef.Table, builder qb.Builder, values ...interface{}) error {
+	syncer.oracleSend(ctx, builder, values...)
+	testRows, err := ds.testStore.load(ctx, builder, values...)
 	if err != nil {
 		return errors.Wrapf(err, "unable to load check data from the test store")
 	}
-	oracleRows, err := ds.oracleStore.load(ctx, builder, values)
+	oracleRows, err := syncer.oracleGet()
 	if err != nil {
 		return errors.Wrapf(err, "unable to load check data from the oracle store")
 	}
```

Comment:
Sorry for confusion, I have glanced into gocql code and figured that it is done so unefficiant that any optimization won't make a dent in pefromance, it is better to just replace this code with simple sync.Waitgroup:
```
	var testRows, oracleRows []map[string]interface{}
	var testErr, oracleErr error
	var wg sync.WaitGroup
	wg.Add(2)
	go func() {
		testRows, testErr = ds.testStore.load(ctx, builder, values)
		wg.Done()
	}()
	go func() {
		oracleRows, oracleErr = ds.oracleStore.load(ctx, builder, values)
		wg.Done()
	}()
	wg.Wait()
```